src/ort_genai_c.cpp

```diff
@@ -246,6 +276,63 @@ OgaResult* OGA_API_CALL OgaTokenizerStreamDecode(OgaTokenizerStream* p, int32_t
   OGA_CATCH
 }
 
+OgaResult* OGA_API_CALL OgaCreateStrings(OgaStrings** out) {
+  OGA_TRY
+  *out = reinterpret_cast<OgaStrings*>(std::make_unique<std::vector<std::string>>().release());
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaCreateAllocatedStrings(size_t num_strings, size_t* str_lengths, OgaStrings** out) {
+  OGA_TRY
+  auto strings = std::make_unique<std::vector<std::string>>(num_strings);
+  for (size_t i = 0; i < num_strings; i++) {
+    strings->at(i).resize(str_lengths[i]);
+  }
+  *out = reinterpret_cast<OgaStrings*>(strings.release());
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaStringsGetBuffer(OgaStrings* string_array, size_t index, char** out) {
+  OGA_TRY
+  *out = &((*reinterpret_cast<std::vector<std::string>*>(string_array))[index])[0];
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaStringsAddString(OgaStrings* string_array, const char* str) {
+  OGA_TRY
+  reinterpret_cast<std::vector<std::string>*>(string_array)->push_back(str);
+  return nullptr;
+  OGA_CATCH
+}
+
+OgaResult* OGA_API_CALL OgaStringsAddStrings(OgaStrings* string_array, const char* const* strings, size_t count) {
```

Comment:
Same thing here, there is not a way to verify that strings is an array of count.

Reply:
Removed OgaStrings type.